File: java_tests/src/main/java/org/infinispan/client/hotrod/configuration/ConnectionPoolConfiguration.java
package org.infinispan.client.hotrod.configuration;

/**
 * ConnectionPoolConfiguration.
 *
 * @author Tristan Tarrant
 * @since 5.3
 */
public class ConnectionPoolConfiguration {
   private ExhaustedAction exhaustedAction = ExhaustedAction.WAIT;
   
   private cli.Infinispan.HotRod.Config.ConnectionPoolConfiguration jniConnectionPoolConfiguration;

   public ConnectionPoolConfiguration(
         cli.Infinispan.HotRod.Config.ConnectionPoolConfiguration jniConnectionPoolConfiguration) {
      this.jniConnectionPoolConfiguration = jniConnectionPoolConfiguration;
      // if (jniConnectionPoolConfiguration != null) {
      //    this.exhaustedAction.setExhaustedAction(this.jniConnectionPoolConfiguration.ExhaustedAction());
      // }
   }
   
   public cli.Infinispan.HotRod.Config.ConnectionPoolConfiguration getJniConnectionPoolConfiguration() {
      return this.jniConnectionPoolConfiguration;
   }

   public ExhaustedAction exhaustedAction() {
      // this.exhaustedAction.setExhaustedAction(this.jniConnectionPoolConfiguration.ExhaustedAction());
      return this.exhaustedAction;
   }

   public boolean lifo() {
      return this.jniConnectionPoolConfiguration.Lifo();
   }

   public int maxActive() {
      return this.jniConnectionPoolConfiguration.MaxActive();
   }

   public int maxTotal() {
      return this.jniConnectionPoolConfiguration.MaxTotal();
   }

   public long maxWait() {
      return this.jniConnectionPoolConfiguration.MaxWait();
   }

   public int maxIdle() {
      return this.jniConnectionPoolConfiguration.MaxIdle();
   }

   public int minIdle() {
      return this.jniConnectionPoolConfiguration.MinIdle();
   }

   public int numTestsPerEvictionRun() {
      return this.jniConnectionPoolConfiguration.NumTestsPerEvictionRun();
   }

   public long timeBetweenEvictionRuns() {
      return this.jniConnectionPoolConfiguration.TimeBetweenEvictionRuns();
   }

   public long minEvictableIdleTime() {
      return this.jniConnectionPoolConfiguration.MinEvictableIdleTime();
   }

   public boolean testOnBorrow() {
      return this.jniConnectionPoolConfiguration.TestOnBorrow();
   }

   public boolean testOnReturn() {
      return this.jniConnectionPoolConfiguration.TestOnReturn();
   }

   public boolean testWhileIdle() {
      return this.jniConnectionPoolConfiguration.TestWhileIdle();
   }

   @Override
   public String toString() {
      return "ConnectionPoolConfiguration [exhaustedAction=" + exhaustedAction() + ", lifo=" + lifo() + ", maxActive=" + maxActive() + ", maxTotal=" + maxTotal() + ", maxWait=" + maxWait()
            + ", maxIdle=" + maxIdle() + ", minIdle=" + minIdle() + ", numTestsPerEvictionRun=" + numTestsPerEvictionRun() + ", timeBetweenEvictionRuns=" + timeBetweenEvictionRuns()
            + ", minEvictableIdleTime=" + minEvictableIdleTime() + ", testOnBorrow=" + testOnBorrow() + ", testOnReturn=" + testOnReturn() + ", testWhileIdle=" + testWhileIdle() + "]";
   }

}

File: src/Infinispan.HotRod/Config/ConnectionPoolConfiguration.cs
namespace Infinispan.HotRod.Config
{
    /// <summary>
    ///   Used to hold connection pool specific configurations.
    /// </summary>
    public class ConnectionPoolConfiguration
    {
        private Infinispan.HotRod.SWIG.ConnectionPoolConfiguration config;

        internal ConnectionPoolConfiguration(Infinispan.HotRod.SWIG.ConnectionPoolConfiguration config)
        {
            this.config = config;
        }

        internal Infinispan.HotRod.SWIG.ConnectionPoolConfiguration Config()
        {
            return config;
        }

        /// <summary>
        ///   Retrieves the configured exhaust action.
        /// </summary>
        public ExhaustedAction ExhaustedAction()
        {
            return config.ExhaustedAction();
        }

        /// <summary>
        ///   Retrieves the state of lifo flag.
        /// </summary>
        public bool Lifo()
        {
            return config.isLifo();
        }
        
        /// <summary>
        ///   Retrieves the configured max active value.
        /// </summary>
        public int MaxActive()
        {
            return config.getMaxActive();
        }
    
        /// <summary>
        ///   Retrieves the configured max total value.
        /// </summary>
        public int MaxTotal()
        {
            return config.getMaxTotal();
        }
    
        /// <summary>
        ///   Retrieves the configured max wait value.
        /// </summary>
        public int MaxWait()
        {
            return config.getMaxWait();
        }
    
        /// <summary>
        ///   Retrieves the configured max idle value.
        /// </summary>
        public int MaxIdle()
        {
            return config.getMaxIdle();
        }
 
        /// <summary>
        ///   Retrieves the configured min idle value.
        /// </summary>   
        public int MinIdle()
        {
            return config.getMinIdle();
        }

        /// <summary>
        ///   Retrieves the configured num test per eviction value.
        /// </summary>    
        public int NumTestsPerEvictionRun()
        {
            return config.getNumTestsPerEvictionRun();
        }

        /// <summary>
        ///   Retrieves the configured time between evictions value.
        /// </summary>    
        public int TimeBetweenEvictionRuns()
        {
            return config.getTimeBetweenEvictionRuns();
        }
   
        /// <summary>
        ///   Retrieves the configured min evictable idle time value.
        /// </summary> 
        public int MinEvictableIdleTime()
        {
            return config.getMinEvictableIdleTime();
        }

        /// <summary>
        ///   Retrieves the state of the configured test on borrow flag.
        /// </summary>
        public bool TestOnBorrow()
        {
            return config.isTestOnBorrow();
        }

        /// <summary>
        ///   Retrieves the state of the configured test on run flag.
        /// </summary>
        public bool TestOnReturn()
        {
            return config.isTestOnReturn();
        }
    
        /// <summary>
        ///   Retrieves the state of the configured test while idle flag.
        /// </summary>
        public bool TestWhileIdle()
        {
            return config.isTestWhileIdle();
        }
    }
}